ffi/src/transaction/mod.rs

```diff
@@ -0,0 +1,355 @@
+//! This module holds functionality for managing transactions.
+mod write_context;
+
+use crate::error::{ExternResult, IntoExternResult};
+use crate::handle::Handle;
+use crate::unwrap_and_parse_path_as_url;
+use crate::KernelStringSlice;
+use crate::{DeltaResult, ExternEngine, Table, Url};
+use crate::{ExclusiveEngineData, SharedExternEngine};
+use delta_kernel::transaction::{CommitResult, Transaction};
+use delta_kernel_ffi_macros::handle_descriptor;
+
+/// A handle representing an exclusive transaction on a Delta table.
+///
+/// This struct provides a safe wrapper around the underlying `Transaction` type,
+/// ensuring exclusive access to transaction operations. The transaction can be used
+/// to stage changes and commit them atomically to the Delta table.
+#[handle_descriptor(target=Transaction, mutable=true, sized=true)]
+pub struct ExclusiveTransaction;
+
+/// Start a transaction on the latest snapshot of the table.
+///
+/// # Safety
+///
+/// Caller is responsible for passing valid handles and path pointer.
+#[no_mangle]
+pub unsafe extern "C" fn transaction(
+    path: KernelStringSlice,
+    engine: Handle<SharedExternEngine>,
+) -> ExternResult<Handle<ExclusiveTransaction>> {
+    let url = unsafe { unwrap_and_parse_path_as_url(path) };
+    let engine = unsafe { engine.as_ref() };
+    transaction_impl(url, engine).into_extern_result(&engine)
+}
+
+fn transaction_impl(
+    url: DeltaResult<Url>,
+    extern_engine: &dyn ExternEngine,
+) -> DeltaResult<Handle<ExclusiveTransaction>> {
+    let table = Table::try_from_uri(url?)?;
```

Comment:
sorry - these `Table` APIs are all gone now, but should be a simple shift to `Snapshot` directly. see https://github.com/delta-io/delta-kernel-rs/pull/976